src/modules/job-ingest/job-ingest.cpp

```diff
@@ -0,0 +1,113 @@
+/*****************************************************************************\
+ *  Copyright (c) 2017 Lawrence Livermore National Security, LLC.  Produced at
+ *  the Lawrence Livermore National Laboratory (cf, AUTHORS, DISCLAIMER.LLNS).
+ *  LLNL-CODE-658032 All rights reserved.
+ *
+ *  This file is part of the Flux resource manager framework.
+ *  For details, see https://github.com/flux-framework.
+ *
+ *  This program is free software; you can redistribute it and/or modify it
+ *  under the terms of the GNU General Public License as published by the Free
+ *  Software Foundation; either version 2 of the license, or (at your option)
+ *  any later version.
+ *
+ *  Flux is distributed in the hope that it will be useful, but WITHOUT
+ *  ANY WARRANTY; without even the IMPLIED WARRANTY OF MERCHANTABILITY or
+ *  FITNESS FOR A PARTICULAR PURPOSE.  See the terms and conditions of the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
+ *  See also:  http://www.gnu.org/licenses/
+\*****************************************************************************/
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <flux/jobspec.hpp>
+
+using namespace Flux::Jobspec;
+
+extern "C" {
+#include <flux/core.h>
```

Comment:
This is not for your code @morrone but since we will increaingly use C++, it seems it would be useful to add __cplusplus support into flux header files.

Reply:
Yes!  Actually, that is exactly what I tried to do.  I went through and made the change to all the C headers in flux core.  Unfortunately, our python interface auto-generator threw a fit when it hit those statements.  It seemed like it wasn't properly preprocessing the files and things were still in its file that should have been removed at preprocessing time.   It looked like it was going to take a while to figure out how to make it happy, so I backtracked and took the easy route.

But yes, that would be the better approach in the long run.